src/Logging/Logging.File/FileLogService.cs

```diff
@@ -1,45 +1,102 @@
 
-        public bool IsEnabledFor(LogLevel logLevel) => _provider.IsEnabledFor(logLevel);
+        protected override async Task PublishMessageBatch(IEnumerable<LogMessage> messageBatch)
+        {
+            Directory.CreateDirectory(Options.Directory);
+
+            foreach (var message in messageBatch)
+            {
+                try
+                {
+                    var path = CreateLogFilePath();
 
-        public void Log<TState>(
-            DateTimeOffset timestamp,
-            LogLevel logLevel,
-            TState state,
-            Exception ex,
-            Func<TState, Exception, string> formatter)
+                    await System.IO.File.AppendAllTextAsync(
+                        path,
+                        MessageFormatter.Format(
+                            message,
+                            Options.TimeStampFormat));
+                }
+                catch { }
+            }
+
+            ClearObsoleteFiles();
+        }
+
+        private string CreateLogFilePath()
         {
-            NotNull(formatter, nameof(formatter));
-
-            _provider.ProcessMessage(new LogMessage(
-                timestamp,
-                logLevel,
-                formatter(state, ex),
-                _context,
-                ex));
+            var path = GetFullLogFilePath();
+
+            if (Options.MaxFileSize == null)
+                return path;
+
+            var size = GetFileSize(path);
+
+            if (size <= Options.MaxFileSize)
+                return path;
+
+            _index++;
+            return GetFullLogFilePath();
+        }
+
+        private string GetFullLogFilePath()
+        {
+            var suffix = GetFileNameSuffix();
+
+            return Path.Combine(
+                Options.Directory,
+                $"{Options.FileNamePrefix}{suffix}.{Options.Extension}");
         }
 
-        public void Log<TState>(
-            LogLevel logLevel,
-            TState state,
-            Exception ex,
-            Func<TState, Exception, string> formatter) => Log(DateTimeOffset.Now, logLevel, state, ex, formatter);
+        private long GetFileSize(string path)
+        {
+            var fi = new FileInfo(path);
+            var size = fi.Exists ?
+                fi.Length :
+                0;
+
+            return size;
+        }
+
+        private string GetFileNameSuffix()
+        {
+            var suffix = _index == 0 ?
+                 string.Empty :
+                 $"-{_index}";
+
+            return suffix;
+        }
+
+        private void ClearObsoleteFiles()
+        {
+            if (Options.MaxFilesToRetain == null)
+                return;
+
+            var filesToRemove = new DirectoryInfo(Options.Directory)
+                .GetFiles(Options.FileNamePrefix + "*." + Options.Extension)
+                .OrderByDescending(f => f.LastWriteTime)
+                .Skip(Options.MaxFilesToRetain.Value);
+
+            foreach (var file in filesToRemove)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'foreach' block.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5626772)